### PR TITLE
[all] Bumped up avro-util dep to pick up a fix in fast serializer generation

### PR DESCRIPTION
**Pull request by gaojieliu**
The latest version of fast-avro added the support of splitting large methods in fast serializer.

## How was this PR tested?
CI

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Review by nisargthakkar — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -39,7 +39,7 @@ if (project.hasProperty('overrideBuildEnvironment')) {
 }
 
 def avroVersion = '1.10.2'
-def avroUtilVersion = '0.2.150'
+def avroUtilVersion = '0.3.17'
```
Comment (FelixGV) [inline build.gradle:42]: That looks like a big bump! We should look carefully to get confidence that there are no regressions....

**Review by FelixGV — APPROVED** [review]
Thanks!

File: build.gradle (hunk 1 above)
Comment (gaojieliu) [reply build.gradle:42]: The CI passed, and we will certainly bake this in EI.
Fast-Avro hasn't updated much, and mostly the avro compatiblity layer.

Comment (nisargthakkar) [reply build.gradle:42]: It's not really too big of a jump in production, actually. Our internal dependencies were bumping up `avro-util` transitively to a much more recent version (`0.3.14`)

Comment (nisargthakkar) [reply build.gradle:42]: There is one regression related to `RandomRecordGenerator` that we've identified - [This PR](https://github.com/linkedin/avro-util/pull/495/files) breaks random object generation for non-record schemas

**Pull request merged**